horace_core/sqw/@sqw/get_nearest_pixels.m

```diff
@@ -92,27 +87,32 @@
 end
 
 % Get coordinates of points along each projection axis
-step = (win.data.axes.img_range(2,:)-win.data.axes.img_range(1,:))./win.data.axes.nbins_all_dims;
-ustep = step(pax);
+step = win.data.axes.get_bin_step()';
+ustep = step(pax)';
 xpstep = xp_ok ./ repmat(ustep,size(xp_ok,1),1);
 
-% Get components along projection axes of the pixels in the sqw object
-uprojstep = calculate_uproj_pixels (win, step);   % in units of steps
-uprojstep = uprojstep';
-
 ipix = zeros(numel(ind),1);
 nend = cumsum(npix);
 nbeg = nend - npix + 1;
+
 ucent = zeros(1,4);
 for i=1:numel(ind)
     % Projections for pixels in the bin
-    uprojstep_bin = uprojstep(nbeg(ind(i)):nend(ind(i)),:);
+    curr_pix = win.pix.get_pix_in_ranges(nbeg(ind(i)), npix(ind(i)));
```

Comment:
This will give only one (current) bin. One should consider all bins adjusent to current, and keep in mind the situation if they all empty